Common/include/mpi_structure.inl

```diff
@@ -612,7 +612,7 @@ inline void CBaseMPIWrapper::Scatter(void *sendbuf, int sendcnt, Datatype sendty
 }
 
 inline void CBaseMPIWrapper::Allgatherv(void *sendbuf, int sendcnt, Datatype sendtype,
-                                   void *recvbuf, int recvcnt, int *displs, Datatype recvtype, Comm comm){
+                                   void *recvbuf, int *recvcnt, int *displs, Datatype recvtype, Comm comm){
```

Comment:
This was the problem, the signature was incorrect, I've had some issues (weird segfaults) in the past with AD and allgatherv by the way... not because of this problem, and maybe it was my fault... But anyway just something to keep in the back of your mind if it causes you problems too.